src/indexed_crate.rs

```diff
@@ -211,6 +211,10 @@ pub struct IndexedCrate<'a> {
     /// index: kind of public top-level item -> `IndexMap<Id, &'a Item>` of that kind
     pub(crate) pub_item_kind_index: PubItemKindIndex<'a>,
 
+    /// index: enum id + variant name -> variant, index of variant inside enum.
+    #[allow(clippy::type_complexity)]
```

Comment:
Let's use
```suggestion
    #[expect(clippy::type_complexity)]
```
so that if this ever stops triggering `type_complexity`, we are reminded to remove the suppression.